src/views/CountryRiskWatch/ReturnPeriodTable/index.tsx

```diff
@@ -0,0 +1,199 @@
+
+import i18n from './i18n.json';
+import styles from './styles.module.css';
+
+type GetCountryRisk = paths['/api/v1/country-seasonal/']['get'];
+type CountryRiskResponse = GetCountryRisk['responses']['200']['content']['application/json'];
+type HazardType = components['schemas']['HazardTypeEnum'];
+interface HazardTypeOption {
+    hazard_type: HazardType;
+    hazard_type_display: string;
+}
+interface TransformedReturnPeriodData {
+    frequencyDisplay: string;
+    displacement: number | undefined;
+    exposure: number | undefined;
+    economicLosses: number | undefined;
+}
+
+function hazardTypeKeySelector(option: HazardTypeOption) {
+    return option.hazard_type;
+}
+function hazardTypeLabelSelector(option: HazardTypeOption) {
+    return option.hazard_type_display;
+}
+function returnPeriodKeySelector(option: TransformedReturnPeriodData) {
+    return option.frequencyDisplay;
+}
+
+interface Props {
+    data: CountryRiskResponse[number]['return_period_data'] | undefined;
+}
+
+function ReturnPeriodTable(props: Props) {
+    const { data } = props;
+    const strings = useTranslation(i18n);
+
+    const [hazardType, setHazardType] = useInputState<HazardType>('FL');
+    const hazardOptions = useMemo(
+        () => (
+            unique(
+                data?.map(
+                    (datum) => {
+                        if (isFalsyString(datum.hazard_type) || datum.hazard_type === '') {
+                            return undefined;
+                        }
+
+                        return {
+                            hazard_type: datum.hazard_type,
+                            hazard_type_display: datum.hazard_type_display,
+                        };
+                    },
+                ).filter(isDefined) ?? [],
+                (datum) => datum.hazard_type_display,
+            )
+        ),
+        [data],
+    );
+
+    const columns = [
+        createStringColumn<TransformedReturnPeriodData, string | number>(
+            'frequency',
+            strings.returnPeriodTableReturnPeriodTitle,
+            (item) => item.frequencyDisplay,
+        ),
+        createNumberColumn<TransformedReturnPeriodData, string | number>(
+            'numRiskOfDisplacement',
+            strings.returnPeriodTableReturnPeriodTitle,
+            (item) => item.displacement,
+            {
+                headerInfoTitle: strings.returnPeriodTableDisplacementTitle,
+                headerInfoDescription: resolveToComponent(
+                    strings.returnPeriodTableDisplacementDescription,
+                    {
+                        source: (
+                            <Link
+                                to="https://www.internal-displacement.org/database/global-displacement-risk-model"
+                            >
+                                {strings.returnPeriodTableDisplacementSource}
+                            </Link>
+                        ),
+                    },
+                ),
```

Comment:
Does this not have any performance impact?

Reply:
Let's benchmark it

Reply:
Yes please